### ISSUE TITLE
API Gateway integration tests should use mocked oauth2 server

### ISSUE THREAD

**Issue by strekm** [kyma-project/api-gateway#561]
**Description**

Replace usage of ORY Hydra in API Gateway integration tests with mocked [oauth2 server](https://github.com/navikt/mock-oauth2-server). Custom domain test still should be using SAP IAS and run on main / release branches only. All other integration tests should run also on PRs.

## State of the task

- Ory and IstioJwt test suites are updated to use the OAuth2 mock and are working on Gardener
- On k3d most of the IstioJwt test suite is working, when workaround for Istio is applied (listed below). Only one test that is using Oathkeeper is failing.
- On k3d ORy testsuite is failing, because of issues with Oathkeeper (listed below)
- Upgrade testsuite wasn't updated to use the  OAuth2 mock server, yet.
- I didn't find a good alternative mock-server where the introspect behaviour can be customized. We still have the option to use a simple self implementation.

### Issues with Istio
- when using `oauth2-mock.local.kyma.dev/default/jwks` as JWKS in the APIRule, istiod logged an error, that self-signed certificates are not supported for JWKS.
- Using the service FQDN didn't work, because of a `Connection refused error` in istiod. This might be fixed by a `PeerAuthentication` but this would change the test setup even more.

#### Workaround
- Add a env variable in istiod deployment to enable verification skipping for JWKS `JWKS_RESOLVER_INSECURE_SKIP_VERIFY` to `"true"`

### Issues with Oathkeeper
- Tests that involve Oathkeeper and have a reference to the OAuth2 mock by it's external address (`oauth2-mock.local.kyma.dev/default`) in the APIRule will fail, because Oathkeeper doesn't allow self-signed certificates.
- If we use the service FQDN, this will work for the requests, but the tokens have the issuer set to `https://oauth2-mock.local.kyma.dev/default`, because this is the address that is used for token creation. One of the issues we experience is then, that the introspect endpoint of the OAuth2 mock will then cause a 401 in Oathkeeper. This happens because the OAUth2 mock derives the issuer from the called URL. In this case it is the service FQDN introspect endpoint e.g. http://mock-oauth2-server.namespace.svc.cluster.local/default/introspect. This doesn't match the issuer `https://oauth2-mock.local.kyma.dev/default` in the token.
 
 Branch: https://github.com/triffer/api-gateway/tree/int-test-oauth2-mock

ACs:
- [x] ORY Hydra is not used in any tests
- [x] integrations tests trigger does not change
- [ ] Rebase branch on latest main 
- [ ] Wait for mock-server to be started before starting tests
- [ ] int tests run on k3d with oauth2 server mock
- [x] int tests run on Gardener with oauth2 server mock
- [x] custom domain test run on main and release branches
- [x] custom domain test still use SAP IAS
- [ ] ci-cd document should be updated or input should be provided

**Reasons**

removing dependency to ORY Hydra

**DoD:**
- [ ] provide documentation
~~- [ ] release notes and What's New updates for Kyma customers~~
~~- [ ] provide unit tests~~
~~- [ ] provide integration tests~~
~~- [ ] test on production-like environment~~
~~- [ ] verify resource limits~~
~~- [ ] followup issue~~
~~- [ ] create release and bump in kyma~~
~~- [ ] PR reviewer will verify code coverage and evaluate if it is acceptable~~

**Attachments**
